src/responder/pam/pamsrv_passkey.h

```diff
@@ -36,14 +36,19 @@ enum passkey_user_verification {
     PAM_PASSKEY_VERIFICATION_INVALID
 };
 
-errno_t passkey_local(TALLOC_CTX *mem_ctx,
-                             struct tevent_context *ev,
-                             struct pam_ctx *pam_ctx,
-                             struct pam_auth_req *preq,
-                             struct pam_data *pd);
-errno_t passkey_kerberos(struct pam_ctx *pctx,
-                         struct pam_data *pd,
-                         struct pam_auth_req *preq);
+/* Operations when calling passkey child */
+enum passkey_child_op {
+    PAM_PASSKEY_OP_PREFLIGHT,
+    PAM_PASSKEY_OP_KERBEROS_AUTH,
+    PAM_PASSKEY_OP_LOCAL_AUTH,
+    PAM_PASSKEY_OP_INVALID
+};
+
+struct pam_preflight_data {
```

Comment:
Can we call it `pam_passkey_data`? The idea would be to include all passkey related data in the same structure

Reply:
> Okay what other data should be stored in `pam_passkey_data` ? I still think it makes sense to separate the preflight data somehow to make it clear it is preflight specific data.

Nothing for the moment, but I see a pattern where this data will grow and we'll need to accommodate it.

In any case, this is just an opinion so if you want to keep this as it is I'm also fine with it. We can always refactor later on when the need is there.